examples/generation-based-subscription/src/components/pricing-card.tsx

```diff
@@ -0,0 +1,215 @@
+'use client';
+
+import { useState } from 'react';
+import { Check } from 'lucide-react';
+import { useBilling } from '@flowglad/nextjs';
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card';
+import { Badge } from '@/components/ui/badge';
+import { Button } from '@/components/ui/button';
+import { cn } from '@/lib/utils';
+
+export interface PricingPlan {
+  name: string;
+  description?: string;
+  displayMonthly: string;
+  displayYearly: string;
+  monthlySlug: string;
+  yearlySlug: string;
+  features: string[];
+  isPopular?: boolean;
+}
+
+interface PricingCardProps {
+  plan: PricingPlan;
+  billingPeriod: 'monthly' | 'yearly';
+  isCurrentPlan?: boolean;
+  hideFeatures?: boolean;
+}
+
+/**
+ * PricingCard component displays a single pricing plan
+ */
+export function PricingCard({
+  plan,
+  billingPeriod,
+  isCurrentPlan = false,
+  hideFeatures = false,
+}: PricingCardProps) {
+  const billing = useBilling();
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const periodLabel = billingPeriod === 'monthly' ? '/month' : '/year';
+  const priceSlug =
+    billingPeriod === 'monthly' ? plan.monthlySlug : plan.yearlySlug;
+  const displayPrice =
+    billingPeriod === 'monthly' ? plan.displayMonthly : plan.displayYearly;
+
```

Comment:
would be nice to add a guard before using values from useBilling, eg.
```suggestion
if (!billing.loaded) {
      return <div>Loading...</div>;
    }
if (billing.errors) {
      return <div>Error loading billing data</div>;
    }
```